File: API/Entities/ToDoList.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace API.Entities;

public enum TagEnum {
    Education,
    Emergency,
    Financial,
    Fun,
    General,
    Health,
    Hobby,
    Holidays,
    Social,
    Travel,
    Work,
}

public class ToDoList
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public required string Title { get; set; }
    public string? Description {get; set; }
    public TagEnum? Tag { get; set; }
    public required DateOnly CreateDate{ get; set; } = DateOnly.FromDateTime(DateTime.Now);
    public DateOnly? EndDate { get; set; }
    public bool Completed { get; set; } = false;

    // Navigation property for AppUser
    public required AppUser User { get; set; }

    // Navigation property for Tasks
    public ICollection<UserTask> Tasks { get; } = [];

}

File: client/src/app/_models/toDoList.ts
import { Task } from "./task";

export enum Tag {
    Education,
    Emergency,
    Financial,
    Fun,
    General,
    Health,
    Hobby,
    Holidays,
    Social,
    Travel,
    Work,
}

export const tagPairs: [Tag, string][] = Object.entries(Tag).filter(keyValue => !isNaN(Number(keyValue[0]))).map(rawKeyValue => [Number(rawKeyValue[0]), rawKeyValue[1] as string])

export const tagMap = new Map<Tag, string>(tagPairs);

export interface CreateToDoListParams {
    title: string;
    description?: string;
    tag?: Tag | undefined;
    endDate?: Date;
}

interface OtherToDoListProperties {
    id: number;
    tasks: Task[];
}

interface OtherUpdateToDoListProperties {
    completed: boolean;
}

export type ToDoList = CreateToDoListParams & OtherToDoListProperties & OtherUpdateToDoListProperties;

export type UpdateToDoListParams = Partial<CreateToDoListParams & OtherUpdateToDoListProperties>;